src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -257,8 +487,29 @@ public IndexModel describeIndex(String indexName) throws PineconeException {
         return indexModel;
     }
 
-    public IndexModel configureIndex(String indexName, String podType, Integer replicas) throws
-            PineconeValidationException {
+    /**
+     * Configures an existing pod-based index with new settings.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *     import org.openapitools.client.model.IndexModel;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     // Make a configuration change
+     *     IndexModel index = client.configureIndex("my-index", "p1.x2", 4);
+     *
+     *     // Call describeIndex to see the index status as the change is applied.
+     *     index = client.describeIndex("my-index");
+     * </pre>
+     *
+     * @param indexName the name of the index to configure.
+     * @param podType the new podType for the index. Can be null if not changing the pod type.
```

Comment:
I'd add `"Can be null if not changing the pod type of the base index held in the collection"`